### PR TITLE
Pixel cleanup

### PR DESCRIPTION
**Pull request by zombiezen**
Clean up pixel format related functions

### CODE REVIEW

File: sdl/pixel.go [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+	PixelTypeArrayU32 PixelType = C.SDL_PIXELTYPE_ARRAYU32
+	PixelTypeArrayF16 PixelType = C.SDL_PIXELTYPE_ARRAYF16
+	PixelTypeArrayF32 PixelType = C.SDL_PIXELTYPE_ARRAYF32
+)
+
+type PixelOrder uint32
+
+// Bitmap pixel order, high bit -> low bit.
+const (
+	BitmapOrderNone PixelOrder = iota
```
Comment (adam000) [inline sdl/pixel.go]: All of these consts are also available from SDL http://wiki.libsdl.org/SDL_PixelFormatEnum

File: sdl/window.go [hunk 2]
```diff
@@ -99,29 +93,31 @@ func NewRenderer(window Window, index int, flags uint32) (Renderer, error) {
...
-	var textureFormats []PixelFormat
-	sliceHeader := (*reflect.SliceHeader)((unsafe.Pointer(&textureFormats)))
-	sliceHeader.Cap = TextureFormatsSize
-	sliceHeader.Len = TextureFormatsSize
-	sliceHeader.Data = uintptr(unsafe.Pointer(&info.texture_formats[0]))
-
-	return &RendererInfo{C.GoString(info.name), RendererFlags(info.flags),
-		uint32(info.num_texture_formats), textureFormats, int(info.max_texture_width),
-		int(info.max_texture_height)}, nil
+	formats := make([]PixelFormatEnum, info.num_texture_formats)
```
Comment (adam000) [inline sdl/window.go:110]: Ahh. I knew there was a better way than the one I found. Cool!

Comment (zombiezen) [inline sdl/window.go:110]: Admittedly, this does a copy, but for 256 bytes on an uncommon operation, who cares.

File: sdl/pixel.go (hunk 1 above)
Comment (zombiezen) [inline sdl/pixel.go]: I was wondering if you were going to make me do that. Done.

**Comment by adam000** [comment]
CR :+1:

**Pull request merged**